### ISSUE TITLE
TSQL: L049 breaks code using = alias

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#2708]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Rule L049 breaks exisiting code that uses `=` syntax.

Example:
```sql
select
  name = null
from table
```

becomes
```sql
select
  name is null
from table
```

### Expected Behaviour

The code remains unchanges, as it is correct.

### Observed Behaviour

The assignment `name = null` is changed to the comparison `name is null` which makes no sense in that context.

### How to reproduce

Run `sqlfluff fix` on the above code sample.

### Dialect

T-SQL

### Version

Latest from git `main`

### Configuration

Nothing relevant apart from setting the dialect.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
In your example, you are aliasing the column as `null`. Does that work in T-SQL, given that `null` is a SQL keyword?

**Comment by fdw** [comment]
No, in my example `null` is the value of a column called `name`. Maybe you need the column for a CTE or a `union`, but it's value is always `null`.

**Comment by tunetheweb** [comment]
Seems related to #2552 

We don’t currently support this in SQLFluff so it’s not surprising the rules are getting confused and assuming you mean something else. I think if this support is added to the TSQL dialect and this is seen as an alias rather than an assignment then this rule may not longer trigger. But would be good to test that if/when that support is added to be sure.

T-SQL has many differences to most other SQL dialects 🤷

**Comment by barrywhart** [comment]
@fdw: Can you clarify what the query is doing?

From your expanation:
>in my example null is the value of a column called name. Maybe you need the column for a CTE or a union, but it's value is always null.

it sounds like this is another way of writing:
```
SELECT
    null AS name
```

Can you confirm?

We need to **understand** this query in order to fix the bug. At this point, I'm not yet convinced this **is** a bug. Thanks!

**Comment by fdw** [comment]
> it sounds like this is another way of writing:
> 
> ```
> SELECT
>     null AS name
> ```
> 
> Can you confirm?

Yes, exactly.

> Perhaps you haven't pulled down the latest code and don't have that fix?

I did try on the the then-latest `main`, I think.

But yes, let's wait until #2552 is implemented.

**Comment by barrywhart** [comment]
Oh, I see -- the test query wasn't parsing because of the table being named `table`. It parses fine if the name is something else, e.g.:
```
select
  name = null
from t
```

I'll reopen the issue. Sorry about that!

**Comment by barrywhart** [comment]
PR #2727 fixed this issue. For completeness and to avoid future breakage, I'll go ahead and add a. test to `L049.yml`.
